### ISSUE TITLE
GalleryFragment crashes upon screen rotation

### ISSUE THREAD

**Issue by kirillt** [ARK-Builders/ARK-Navigator#4]
Necessary to save state properly

**Comment by kirillt** [comment]
Caused by #29

**Comment by kirillt** [comment]
It was fixed for a while, but crashes again:
```
V/ActivityThread: Config is null for app:space.taran.arknavigator
E/: [ZeroHung]zrhung_get_config: Get config failed for wp[0x0102]
V/ActivityThread: callActivityOnCreate
D/main: creating
D/AndroidRuntime: Shutting down VM
W/an.arknavigato: Accessing hidden method Lcom/msic/qarth/PatchStore;->createDisableExceptionQarthFile(Ljava/lang/Throwable;)Z (blacklist, JNI)
E/an.arknavigato: [qarth_debug:]  get PatchStore::createDisableExceptionQarthFile method fail.
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: space.taran.arknavigator, PID: 23410
    java.lang.RuntimeException: Unable to start activity ComponentInfo{space.taran.arknavigator/space.taran.arknavigator.ui.activity.MainActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment space.taran.arknavigator.ui.fragments.ResourcesFragment: could not find Fragment constructor
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3430)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3614)
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5527)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5418)
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199)
        at android.os.Handler.dispatchMessage(Handler.java:112)
        at android.os.Looper.loop(Looper.java:216)
        at android.app.ActivityThread.main(ActivityThread.java:7625)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
     Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment space.taran.arknavigator.ui.fragments.ResourcesFragment: could not find Fragment constructor
        at androidx.fragment.app.Fragment.instantiate(Fragment.java:538)
        at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
        at androidx.fragment.app.FragmentManagerImpl$6.instantiate(FragmentManagerImpl.java:2851)
        at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:85)
        at androidx.fragment.app.FragmentManagerImpl.restoreSaveState(FragmentManagerImpl.java:2488)
        at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:195)
        at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:288)
        at androidx.appcompat.app.AppCompatActivity.onCreate(AppCompatActivity.java:115)
        at moxy.MvpAppCompatActivity.onCreate(MvpAppCompatActivity.java:25)
        at space.taran.arknavigator.ui.activity.MainActivity.onCreate(MainActivity.kt:54)
        at android.app.Activity.performCreate(Activity.java:7458)
        at android.app.Activity.performCreate(Activity.java:7448)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1286)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3409)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3614) 
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5527) 
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5418) 
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199) 
        at android.os.Handler.dispatchMessage(Handler.java:112) 
        at android.os.Looper.loop(Looper.java:216) 
        at android.app.ActivityThread.main(ActivityThread.java:7625) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987) 
     Caused by: java.lang.NoSuchMethodException: <init> []
        at java.lang.Class.getConstructor0(Class.java:2327)
        at java.lang.Class.getConstructor(Class.java:1725)
        at androidx.fragment.app.Fragment.instantiate(Fragment.java:523)
        at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57) 
        at androidx.fragment.app.FragmentManagerImpl$6.instantiate(FragmentManagerImpl.java:2851) 
        at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:85) 
        at androidx.fragment.app.FragmentManagerImpl.restoreSaveState(FragmentManagerImpl.java:2488) 
        at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:195) 
        at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:288) 
        at androidx.appcompat.app.AppCompatActivity.onCreate(AppCompatActivity.java:115) 
        at moxy.MvpAppCompatActivity.onCreate(MvpAppCompatActivity.java:25) 
        at space.taran.arknavigator.ui.activity.MainActivity.onCreate(MainActivity.kt:54) 
        at android.app.Activity.performCreate(Activity.java:7458) 
        at android.app.Activity.performCreate(Activity.java:7448) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1286) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3409) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3614) 
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5527) 
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5418) 
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199) 
        at android.os.Handler.dispatchMessage(Handler.java:112) 
        at android.os.Looper.loop(Looper.java:216) 
        at android.app.ActivityThread.main(ActivityThread.java:7625) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987) 
```

**Comment by vedranqa** [comment]
cant reproudce with build 135

**Comment by kirillt** [comment]
@vedranqa this must be reproducible in `main` build (133, 138, 154), I think 135 also should fail when you open a picture and turn phone. What device you tested it with? Build 170 should have fixed this bug.

**Comment by vedranqa** [comment]
Here was my mistake, Build number is 170, where this is checked and I can't reproduce. 
Used Xiaomi A2.

**Comment by kirillt** [comment]
Closed by #135
